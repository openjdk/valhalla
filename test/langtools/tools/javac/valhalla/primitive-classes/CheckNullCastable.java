/*
 * @test /nodynamiccopyright/
 * @summary null cannot be casted to and compared with value types.
 *
 * @compile/fail/ref=CheckNullCastable.out -XDrawDiagnostics CheckNullCastable.java
 */

primitive final class CheckNullCastable {
    void foo(CheckNullCastable cnc) {
        CheckNullCastable cncl = (CheckNullCastable) null;
        if (cnc != null) {};
        if (null != cnc) {};
    }
    int x = 10;
}
