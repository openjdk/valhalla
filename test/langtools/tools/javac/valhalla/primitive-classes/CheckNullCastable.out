CheckNullCastable.java:10:54: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: compiler.misc.type.null, CheckNullCastable)
CheckNullCastable.java:11:17: compiler.err.incomparable.types: CheckNullCastable, compiler.misc.type.null
CheckNullCastable.java:12:18: compiler.err.incomparable.types: compiler.misc.type.null, CheckNullCastable
3 errors
